### PR TITLE
EVM fee proxy

### PR DESCRIPTION
**Pull request by KarishmaBothara**
## Summary

Tested on [porcini](https://portal.rootnet.cloud/?rpc=wss%3A%2F%2Fporcini.rootnet.app%2Farchive%2Fws#/explorer/query/0x68461412b0122f85787e3a6504937ce925c19c1f3688445e633e3e07fb56f43e)

Closes https://centralitydev.atlassian.net/browse/TRN-193
## Checklist

- [x] Add description
- [x] Tag related issue(s)

### CODE REVIEW

**Review by aidan-starke — CHANGES_REQUESTED** [review]
Looks good, can you please run `pnpm lint` and fix warnings, and remove extra examples from README

File: examples/evm/use-feeProxy/src/callWithFeePreferences.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		gasPrice: fees.gasPrice!,
+	};
+
+	await wallet.signTransaction(unsignedTx);
+	const tx = await wallet.sendTransaction(unsignedTx);
+	const receipt = await tx.wait();
+	console.log("receipt::", receipt);
+	// check updated balances
+	const tokenBalanceUpdated = await feeToken.balanceOf(wallet.address);
+	console.log("tokenBalanceUpdated::", tokenBalanceUpdated.toString());
```
Comment (vldolot) [inline examples/evm/use-feeProxy/src/callWithFeePreferences.ts:63]: Remove the double `::` (just use one `:`) to be consistent with the other examples' console.log style.

File: packages/utils/src/getAmountIn.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	const fees = await provider.getFeeData();
+	const txCostXRP = estimate
+		.mul(fees.gasPrice!)
+		.div(10 ** 12)
+		.toNumber();
+	const result = await provider.send("dex_getAmountsIn", [
+		txCostXRP,
+		[feeTokenAssetId, GAS_TOKEN_ID],
+	]);
+	console.log("result::", result);
```
Comment (vldolot) [inline packages/utils/src/getAmountIn.ts:18]: Same here.

File: examples/evm/use-feeProxy/src/callWithFeePreferences.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		to: FEE_PROXY_ADDRESS,
+		nonce: nonce,
+		data: feeProxy.interface.encodeFunctionData("callWithFeePreferences", [
+			feeToken.address,
+			maxFeePaymentInToken,
+			feeToken.address,
+			transferInput,
+		]),
+		gasLimit: gasEstimate,
+		gasPrice: fees.gasPrice!,
```
Comment (vldolot) [inline examples/evm/use-feeProxy/src/callWithFeePreferences.ts:54]: Just a suggestion:
Can put the values for `data`, `gasLimit` and `gasPrice` into its own variable with the same name so you can do tha property shorthand, e.g.:
```js
const unsignedTx = {
  ...,
  nonce,
  data,
  gasLimit,
  gasPrice
};
```

**Review by aidan-starke — APPROVED** [review]

**Pull request merged**